Model/Service/Index.php

```diff
@@ -144,6 +149,7 @@ public function __construct(
         TimezoneInterface $magentoTimeZone,
         NostoDataHelper $nostoDataHelper,
         Sync $nostoSyncService,
+        SyncBulkPublisher $syncBulkPublisher,
```

Comment:
Let's use interface (for example `SyncBulkPublisherInterface`) here and implement do DI configuration to inject the default `SyncBulkPublisher`. `SyncBulkPublisher` would implement the interface. I believe we only need one method `publishCollectionToQueue` or let's call it simply `publishCollectionToQueue`